### ISSUE TITLE
Query: projected collection navigations don't get tracked if the collection is composed on

### ISSUE THREAD

**Issue by maumar** [aspnet/EntityFrameworkCore#8999]
Related to #7131

We currently preform fixup if the collection is projected direcly (by using include pipeline), however if it's composed on, we don't.

e.g.:

```cs
context.Parents.Select(p => new {Parent = p, p.Children.Where(c => c.IsNaughty)}).ToList();
```

We should either change the way we do fixup for those cases, or improve the logic that converts collection projections to includes to incorporate more cases (esp. filters)

**Comment by julielerman** [comment]
Bad news...I thought this was fixed but with 2.0.1, I'm back to this old problem.
In this query:

```
 _context.Samurais.Select(s =>
                   new { Samurai = s, Quotes = s.Quotes })
                   .ToList();
```

The resulting object's Samurai is aware of its quotes. 

But when I filter the quotes ala

```
 _context.Samurais.Select(s =>
                   new { Samurai = s, Quotes = s.Quotes.Where(q => q.Text.Contains("happy")) })
                   .ToList();
```

The object's quote property has the quotes, but it's samurai's quotes property is empty.

Even if I explicitly append ToList() to the subquery

```
 var happySamurais = _context.Samurais.Select(s =>
                   new { Samurai = s, Quotes = s.Quotes.Where(q => q.Text.Contains("happy")).ToList() })
                   .ToList();
```

The fixup isn't happening.

**Comment by julielerman** [comment]
🎉
